ITS GPU: First step of ITS GPU code reorganization

mconcas opened the pull request:
- CUDA vertexer works again
- HIP vertexer works again
- Single headers used for both GPU libraries 
- `.hip.cxx` files generated at build time

mconcas committed:
Add ITS GPU vertexing refactorization

Fix discarded [[ nodiscard ]] warnings

Add self-generation for Vertexer code

davidrohr commented:
@mconcas : Can you fix the codechecker warnings:
```
/sw/SOURCES/O2/6996-slc8_x86-64/0/Detectors/ITSMFT/ITS/tracking/GPU/ITStrackingGPU/VertexerTraitsGPU.h:51:3: error: annotate this function with 'override' or (rarely) 'final' [modernize-use-override]
/sw/SOURCES/O2/6996-slc8_x86-64/0/Detectors/ITSMFT/ITS/tracking/GPU/ITStrackingGPU/VertexerTraitsGPU.h:51:3: error: annotate this function with 'override' or (rarely) 'final' [modernize-use-override]
/sw/SOURCES/O2/6996-slc8_x86-64/0/Detectors/ITSMFT/ITS/tracking/GPU/ITStrackingGPU/VertexerTraitsGPU.h:51:3: error: annotate this function with 'override' or (rarely) 'final' [modernize-use-override]
/sw/SOURCES/O2/6996-slc8_x86-64/0/Detectors/ITSMFT/ITS/tracking/GPU/ITStrackingGPU/VertexerTraitsGPU.h:51:3: error: annotate this function with 'override' or (rarely) 'final' [modernize-use-override
```
I will have a closer look once it passes the FullCI

mconcas commented:
Ok, I'll do it, but I will have to rebase now because of some large pr on its tracking.
It's Mostly cosmetics, btw, for the moment.

mconcas committed:
Fix test macro

mconcas committed:
Fix testing of macro

mconcas commented:
@davidrohr : rebased this on latest changes on ITS tracking.
Hopefully I also fixed the red tests.
From now on this can be considered as "willing to merge".
